src/wp-includes/fonts/class-wp-font-family.php

```diff
@@ -0,0 +1,615 @@
+				$this->data['slug'],
+				$font_face,
+				$src,
+				$suffix
+			);
+			$new_src  = $this->download_asset( $src, $filename );
+			if ( $new_src ) {
+				$new_font_face['src'][] = $new_src;
+			}
+		}
+
+		if ( count( $new_font_face['src'] ) === 1 ) {
+			$new_font_face['src'] = $new_font_face['src'][0];
+		}
+
+		// Remove the download url reference from the font face definition
+		// because it is no longer needed.
+		unset( $new_font_face['downloadFromUrl'] );
+
+		return $new_font_face;
+	}
+
+
+	/**
+	 * Downloads font face assets if the font family is a Google font,
+	 * or moves them if it is a local font.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param array $files An array of files to be installed.
+	 * @return bool True if the font faces were downloaded or moved successfully, false otherwise.
+	 */
+	private function download_or_move_font_faces( $files ) {
+		if ( ! $this->has_font_faces() ) {
+			return true;
+		}
+
+		$new_font_faces = array();
+		foreach ( $this->data['fontFace'] as $font_face ) {
+			// If the fonts are not meant to be dowloaded or uploaded
+			// (for example to install fonts that use a remote url).
+			$new_font_face = $font_face;
+
+			$font_face_is_repeated = false;
+
+			// If the font face has the same fontStyle and fontWeight as an existing, continue.
+			foreach ( $new_font_faces as $font_to_compare ) {
+				if ( $new_font_face['fontStyle'] === $font_to_compare['fontStyle'] &&
+					$new_font_face['fontWeight'] === $font_to_compare['fontWeight'] ) {
+					$font_face_is_repeated = true;
+				}
+			}
+
+			if ( $font_face_is_repeated ) {
+				continue;
+			}
+
+			// If installing google fonts, download the font face assets.
+			if ( ! empty( $font_face['downloadFromUrl'] ) ) {
+				$new_font_face = $this->download_font_face_assets( $new_font_face );
+			}
+
+			// If installing local fonts, move the font face assets from
+			// the temp folder to the wp fonts directory.
+			if ( ! empty( $font_face['uploadedFile'] ) && ! empty( $files ) ) {
+				$new_font_face = $this->move_font_face_asset(
+					$new_font_face,
+					$files[ $new_font_face['uploadedFile'] ]
+				);
+			}
+
+			/*
+			 * If the font face assets were successfully downloaded, add the font face
+			 * to the new font. Font faces with failed downloads are not added to the
+			 * new font.
+			 */
+			if ( ! empty( $new_font_face['src'] ) ) {
+				$new_font_faces[] = $new_font_face;
+			}
+		}
+
+		if ( ! empty( $new_font_faces ) ) {
+			$this->data['fontFace'] = $new_font_faces;
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Gets the post for a font family.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @return WP_Post|null The post for this font family object or
+	 *                      null if the post does not exist.
+	 */
+	public function get_font_post() {
+		$args = array(
+			'post_type'      => 'wp_font_family',
+			'post_name'      => $this->data['slug'],
+			'name'           => $this->data['slug'],
```

Comment:
Do you need post of these?

Do we need to prime post meta and term caches?

Reply:
> Do you need post of these?

@spacedmonkey I don't know what you mean, can you add more context, please?

Reply:
The `wp_font_family` posts are not using metas.